parquet-thrift/src/main/java/parquet/thrift/BufferedProtocolReadToWrite.java

```diff
@@ -358,11 +361,28 @@ public String toDebugString() {
       final TField currentField = field;
       ThriftField expectedField;
       if ((expectedField = type.getChildById(field.id)) == null) {
-        notifyIgnoredFieldsOfRecord(field);
-        hasFieldsIgnored |= true;
-        //read the value and ignore it, NullProtocol will do nothing
-        new ProtocolReadToWrite().readOneValue(in, new NullProtocol(), field.type);
-        continue;
+
+        switch (type.isUnion()) {
+          case NO:
+            // this is an unrecognized field in a struct, not a union
+            notifyIgnoredFieldsOfRecord(field);
+            hasFieldsIgnored |= true;
+            //read the value and ignore it, NullProtocol will do nothing
+            new ProtocolReadToWrite().readOneValue(in, new NullProtocol(), field.type);
+            continue;
+          case YES:
+            // this is a union with an unrecognized member -- this is fatal for this record
+            // in the write path, because it will be unreadable in the read path.
+            // throwing here means we will skip this record entirely.
```

Comment:
You mean it depends on the configuration? Some modes fail on the first bad record?